src/main/java/com/github/mkopylec/charon/core/http/RequestForwarder.java

```diff
@@ -76,11 +78,38 @@ public RequestForwarder(
         traceInterceptor.onForwardComplete(traceId, response.getStatus(), response.getBody(), response.getHeaders());
         receivedResponseInterceptor.intercept(response);
 
+        prepareForwardedResponseHeaders(response);
+
         return status(response.getStatus())
                 .headers(response.getHeaders())
                 .body(response.getBody());
     }
 
+    /**
+     * Remove any protocol-level headers from the remote server's response that
+     * do not apply to the new response we are sending.
+     */
+    private void prepareForwardedResponseHeaders(ResponseData response) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.putAll(response.getHeaders());
+        response.setHeaders(headers);
+
+        headers.remove("Transfer-Encoding");
```

Comment:
Remote servers which use `Chunked` transfer encoding would fail to proxy correctly because line 85 causes Spring to buffer the remote body into a single entity, but we were previously passing over the remote `Transfer-Encoding: Chunked` header, confusing clients.

This fixes that issue. I have tested against a remote server with Chunked encoding, it works after this change but did not work before this change.

I have not added a unit test for this fix.